### PR TITLE
feat: istioctl experimental remove-from-mesh deployment The error message should be better

### PR DESCRIPTION
**Pull request by tanjunchen**
…sage should be better

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by tanjunchen** [comment]
ref:https://github.com/istio/istio/issues/25860

File: istioctl/cmd/add-to-mesh.go [hunk 1]
```diff
@@ -316,6 +316,10 @@ func injectSideCarIntoDeployment(client kubernetes.Interface, deps []appsv1.Depl
 	for _, dep := range deps {
 		log.Debugf("updating deployment %s.%s with Istio sidecar injected",
 			dep.Name, dep.Namespace)
+		if dep.Name == istioEgressgateway || dep.Name == istioIngressgateway {
+			fmt.Fprintf(writer, "failed to inject sidecar to deployment resource %s.%s. Skipping.\n", dep.Name, dep.Namespace)
```
Comment (shamsher31) [inline istioctl/cmd/add-to-mesh.go:320]: I think using `multierror.Append` will be consistent here ?
https://github.com/istio/istio/blob/e1e8825bc02a7aab683a37c02790d6bc3a3a0a9c/istioctl/cmd/add-to-mesh.go#L325-L326

File: istioctl/cmd/remove-from-mesh.go [hunk 2]
```diff
@@ -187,6 +187,10 @@ func unInjectSideCarFromDeployment(client kubernetes.Interface, deps []appsv1.De
 			dep.Name, dep.Namespace)
 		res := dep.DeepCopy()
 		depName := strings.Join([]string{dep.Name, dep.Namespace}, ".")
+		if dep.Name == istioEgressgateway || dep.Name == istioIngressgateway {
+			fmt.Fprintf(writer, "failed to remove the deployment %q from the mesh. remove-from-mesh just removes the sidecar container. Skipping.\n", depName)
```
Comment (shamsher31) [inline istioctl/cmd/remove-from-mesh.go:191]: Can also `multierror.Append` here.

Comment (tanjunchen) [reply istioctl/cmd/remove-from-mesh.go:191]: I think so. Thanks for your comments. Let us consider  @howardjohn  point of view.

Comment (tanjunchen) [reply istioctl/cmd/remove-from-mesh.go:191]: see https://github.com/istio/istio/issues/25860#issuecomment-664089111

File: istioctl/cmd/add-to-mesh.go [hunk 3]
```diff
@@ -316,6 +316,10 @@ func injectSideCarIntoDeployment(client kubernetes.Interface, deps []appsv1.Depl
 	for _, dep := range deps {
 		log.Debugf("updating deployment %s.%s with Istio sidecar injected",
 			dep.Name, dep.Namespace)
+		if dep.Name == istioEgressgateway || dep.Name == istioIngressgateway {
```
Comment (howardjohn) [inline istioctl/cmd/add-to-mesh.go:319]: These names are not special, they just happen to be used in default install. Please avoid special logic based on these names. Prefer a check that looks for istio-proxy container?

**Comment by tanjunchen** [comment]
/close
due to the similar ps https://github.com/istio/istio/pull/25887

**Pull request closed without merging**